packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -510,11 +510,18 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
     }
 
     // Adjust the clip rect for this sliver by the overlap from the previous sliver.
+    final double startOfOverlap = child.constraints.viewportMainAxisExtent - child.constraints.remainingPaintExtent;
+    // We tried to subtract infinity from infinity and got NaN.
+    // This can happen when a RenderShrinkWrappingViewport is given infinite
+    // constraints.
+    if (startOfOverlap.isNaN) {
```

Comment:
We should just check in the existing if above if child.constraints.viewportMainAxisExtent is infinite and then bail out there.

Reply:
Done